File: online-shop/src/app/_model/item-data-presentation.ts
import { Item } from "app/_model/item";
import { Color } from "app/_model/color";
import { ItemData } from "app/_model/item-data";
import { Size } from "app/_model/size";

export class ItemDataPresentation {

    private _item: Item;

    private _colors: Color[];
    private _sizes: Size[];
    private _cost: number;

    private _isNew: boolean;

    public get item(): Item {
        return this._item;
    }
    public get colors(): Color[] {
        return this._colors;
    }
    public get sizes(): Size[] {
        return this._sizes;
    }
    public get cost(): number {
        return this._cost
    }
    public get isNew(): boolean {
        return this._isNew;
    }

    constructor(item: Item, colors: Color[], sizes: Size[], cost: number, isNew: boolean) {
        this._item = item;
        this._colors = colors;
        this._sizes = sizes;
        this._cost = cost;
        this._isNew = isNew;
    }


    public static fromJSON(itemData: any): ItemDataPresentation {
        return new ItemDataPresentation(
            Item.fromJSON(itemData.item),
            itemData.colors.map(color => Color.fromJSON(color)),
            itemData.sizes.map(size => Size.fromJSON(size)),
            itemData.cost,
            itemData.isNew
        );
    } 
    
}

File: online-shop-netbeans-project/src/main/java/main/objects/ItemDataPresentation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package main.objects;

import java.sql.Date;
import java.util.Set;

/**
 *
 * @author admin
 */
public class ItemDataPresentation {

    private Item item;
    private Set<String> colors;
    private Set<String> sizes;
    private int cost;
    private boolean isNew;

    public ItemDataPresentation() {
    }

    public void setItem(Item item) {
        this.item = item;
    }

    public Item getItem() {
        return item;
    }

    public void setColors(Set<String> colors) {
        this.colors = colors;
    }

    public void setSizes(Set<String> sizes) {
        this.sizes = sizes;
    }

    public void setCost(int cost) {
        this.cost = cost;
    }

    public void setIsNew(Date date) {
        this.isNew = true;
    }

}
